pkg/topo/create.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package topo
+
+import (
+	"bytes"
+	"context"
+	topoapi "github.com/onosproject/onos-api/go/onos/topo"
+	"github.com/onosproject/onos-lib-go/pkg/cli"
+	"github.com/spf13/cobra"
+	"time"
+)
+
+func getCreateEntityCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "entity <id> [args]",
+		Args:  cobra.MinimumNArgs(1),
+		Short: "Create Entity",
+		RunE:  runCreateEntityCommand,
+	}
+	cmd.Flags().StringP("kind", "k", "", "Kind ID")
+	//_ = cmd.MarkFlagRequired("kind")
+	cmd.Flags().StringToStringP("aspect", "a", map[string]string{}, "aspect of this entity")
+	return cmd
+}
+
+func getCreateRelationCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "relation <id> <src-entity-id> <tgt-entity-id> [args]",
+		Args:  cobra.MinimumNArgs(3),
+		Short: "Create Relation",
+		RunE:  runCreateRelationCommand,
+	}
+	cmd.Flags().StringP("kind", "k", "", "Kind ID")
+	//_ = cmd.MarkFlagRequired("kind")
+	cmd.Flags().StringToStringP("aspect", "a", map[string]string{}, "aspect of this relation")
+	return cmd
+}
+
+func getCreateKindCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "kind <id> <name> [args]",
+		Args:  cobra.MinimumNArgs(2),
+		Short: "Create Kind",
+		RunE:  runCreateKindCommand,
+	}
+	cmd.Flags().StringToStringP("aspects", "a", map[string]string{}, "default aspect for entities of this kind")
```

Comment:
Is the value here expected to be a raw JSON blob?

Reply:
Yes.

Reply:
It should be "aspect" though; typo.

Reply:
I can also add the --set type.attribute=value usage as with update.

Reply:
Yes I agree it should be singular since each instance of the flag is a single aspect